homeassistant/components/cloud/backup.py

```diff
@@ -152,6 +155,13 @@ async def async_upload_backup(
                     ) from err
                 raise BackupAgentError(f"Failed to upload backup {err}") from err
             except CloudError as err:
+                if (
+                    isinstance(err, CloudApiError)
+                    and isinstance(err.orig_exc, ClientResponseError)
+                    and err.orig_exc.status == HTTPStatus.FORBIDDEN
+                    and self._cloud.subscription_expired
```

Comment:
We may want to move this check to the client library and raise `CloudApiNonRetryableError` with an appropriate error message instead. But I think it's ok to add this here for now.

Reply:
https://github.com/NabuCasa/hass-nabucasa/pull/834